File: magellan-legacy/src/main/java/com/wealthfront/magellan/BaseScreenView.java
package com.wealthfront.magellan;

import android.view.View;
import android.content.Context;
import android.widget.FrameLayout;

import org.jetbrains.annotations.NotNull;

/**
 * Base class to easily implement a {@link ScreenView}. Inherit from {@link FrameLayout}.
 */
public class BaseScreenView<S extends Screen> extends FrameLayout implements ScreenView<S> {

  private S screen;

  public BaseScreenView(@NotNull Context context) {
    super(context);
  }

  @Override
  public final void setScreen(@NotNull S screen) {
    this.screen = screen;
  }

  @Override
  @NotNull
  public final S getScreen() {
    return screen;
  }

  @NotNull
  public View inflate(int resource) {
    return inflate(getContext(), resource, this);
  }
}

File: magellan-kotlin/src/main/java/com/wealthfront/magellan/kotlin/BaseScreenView.kt
package com.wealthfront.magellan.kotlin

import android.content.Context
import android.view.View
import android.widget.FrameLayout

/**
 * Base class to easily implement a [ScreenView]. Inherit from [FrameLayout].
 */
class BaseScreenView<S>(context: Context) : FrameLayout(context), ScreenView<S> where S : Screen<*> {

  private lateinit var screen: S

  override fun setScreen(screen: S) {
    this.screen = screen
  }

  override fun getScreen(): S = screen

  fun inflate(resource: Int): View = View.inflate(context, resource, this)

}
